pandas/io/json/_json.py

```diff
@@ -145,75 +147,51 @@ def _format_axes(self):
         raise AbstractMethodError(self)
 
     def write(self):
-        return self._write(
-            self.obj,
-            self.orient,
-            self.double_precision,
-            self.ensure_ascii,
-            self.date_unit,
-            self.date_format == "iso",
-            self.default_handler,
-            self.indent,
-        )
-
-    def _write(
-        self,
-        obj,
-        orient: Optional[str],
-        double_precision: int,
-        ensure_ascii: bool,
-        date_unit: str,
-        iso_dates: bool,
-        default_handler: Optional[Callable[[Any], JSONSerializable]],
-        indent: int,
-    ):
+        iso_dates = self.date_format == "iso"
         return dumps(
-            obj,
-            orient=orient,
-            double_precision=double_precision,
-            ensure_ascii=ensure_ascii,
-            date_unit=date_unit,
+            self.obj_to_write,
+            orient=self.orient,
+            double_precision=self.double_precision,
+            ensure_ascii=self.ensure_ascii,
+            date_unit=self.date_unit,
             iso_dates=iso_dates,
-            default_handler=default_handler,
-            indent=indent,
+            default_handler=self.default_handler,
+            indent=self.indent,
         )
 
+    @property
+    @abstractmethod
+    def obj_to_write(self) -> Union[NDFrame, Dict[str, Any]]:
+        """Object to write in JSON format."""
+
```

Comment:
can you explicitly `pass` here

are the keys in the Dict necessarily str, or can they be any Index entries?

Reply:
Oh yes, actually the keys in the dictionary can be of ``IndexLabel`` type.
In ``FrameWriter`` the resulting dictionary keys can be tuples of strings in case of multicolumns.
In other cases the keys seems to be only strings.